// llama-stack-client/resources/inference/inference.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as EmbeddingsAPI from "./embeddings.mjs";
export class Inference extends APIResource {
    constructor() {
        super(...arguments);
        this.embeddings = new EmbeddingsAPI.Embeddings(this._client);
    }
    chatCompletion(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/inference/chat_completion', {
            body,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
            stream: params.stream ?? false,
        });
    }
    completion(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/inference/completion', {
            body,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
}
(function (Inference) {
    Inference.Embeddings = EmbeddingsAPI.Embeddings;
})(Inference || (Inference = {}));
