// llama-stack-client/resources/memory/memory.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Memory = void 0;
const resource_1 = require("../../resource.js");
const core_1 = require("../../core.js");
const DocumentsAPI = __importStar(require("./documents.js"));
class Memory extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.documents = new DocumentsAPI.Documents(this._client);
    }
    create(params, options) {
        const { body, 'X-LlamaStack-ProviderData': xLlamaStackProviderData } = params;
        return this._client.post('/memory/create', {
            body: body,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    retrieve(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...query } = params;
        return this._client.get('/memory/get', {
            query,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    update(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/memory/update', {
            body,
            ...options,
            headers: {
                Accept: '*/*',
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    list(params = {}, options) {
        if ((0, core_1.isRequestOptions)(params)) {
            return this.list({}, params);
        }
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData } = params;
        return this._client.get('/memory/list', {
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    drop(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/memory/drop', {
            body,
            ...options,
            headers: {
                Accept: 'application/json',
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    insert(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/memory/insert', {
            body,
            ...options,
            headers: {
                Accept: '*/*',
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    query(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/memory/query', {
            body,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
}
exports.Memory = Memory;
(function (Memory) {
    Memory.Documents = DocumentsAPI.Documents;
})(Memory = exports.Memory || (exports.Memory = {}));
