// llama-stack-client/index.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var _a;
import * as Errors from "./error.mjs";
import * as Uploads from "./uploads.mjs";
import * as Core from "./core.mjs";
import * as API from "./resources/index.mjs";
const environments = {
    production: 'http://any-hosted-llama-stack-client.com',
    sandbox: 'https://example.com',
};
/**
 * API Client for interfacing with the Llama Stack Client API.
 */
export class LlamaStackClient extends Core.APIClient {
    /**
     * API Client for interfacing with the Llama Stack Client API.
     *
     * @param {Environment} [opts.environment=production] - Specifies the environment URL to use for the API.
     * @param {string} [opts.baseURL=process.env['LLAMA_STACK_CLIENT_BASE_URL'] ?? http://any-hosted-llama-stack-client.com] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('LLAMA_STACK_CLIENT_BASE_URL'), ...opts } = {}) {
        const options = {
            ...opts,
            baseURL,
            environment: opts.environment ?? 'production',
        };
        if (baseURL && opts.environment) {
            throw new Errors.LlamaStackClientError('Ambiguous URL; The `baseURL` option (or LLAMA_STACK_CLIENT_BASE_URL env var) and the `environment` option are given. If you want to use the environment you must pass baseURL: null');
        }
        super({
            baseURL: options.baseURL || environments[options.environment || 'production'],
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        this.telemetry = new API.Telemetry(this);
        this.agents = new API.Agents(this);
        this.datasets = new API.Datasets(this);
        this.evaluate = new API.Evaluate(this);
        this.evaluations = new API.Evaluations(this);
        this.inference = new API.Inference(this);
        this.safety = new API.Safety(this);
        this.memory = new API.Memory(this);
        this.postTraining = new API.PostTraining(this);
        this.rewardScoring = new API.RewardScoringResource(this);
        this.syntheticDataGeneration = new API.SyntheticDataGenerationResource(this);
        this.batchInference = new API.BatchInference(this);
        this.models = new API.Models(this);
        this.memoryBanks = new API.MemoryBanks(this);
        this.shields = new API.Shields(this);
        this._options = options;
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
}
_a = LlamaStackClient;
LlamaStackClient.LlamaStackClient = _a;
LlamaStackClient.DEFAULT_TIMEOUT = 60000; // 1 minute
LlamaStackClient.LlamaStackClientError = Errors.LlamaStackClientError;
LlamaStackClient.APIError = Errors.APIError;
LlamaStackClient.APIConnectionError = Errors.APIConnectionError;
LlamaStackClient.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
LlamaStackClient.APIUserAbortError = Errors.APIUserAbortError;
LlamaStackClient.NotFoundError = Errors.NotFoundError;
LlamaStackClient.ConflictError = Errors.ConflictError;
LlamaStackClient.RateLimitError = Errors.RateLimitError;
LlamaStackClient.BadRequestError = Errors.BadRequestError;
LlamaStackClient.AuthenticationError = Errors.AuthenticationError;
LlamaStackClient.InternalServerError = Errors.InternalServerError;
LlamaStackClient.PermissionDeniedError = Errors.PermissionDeniedError;
LlamaStackClient.UnprocessableEntityError = Errors.UnprocessableEntityError;
LlamaStackClient.toFile = Uploads.toFile;
LlamaStackClient.fileFromPath = Uploads.fileFromPath;
export const { LlamaStackClientError, APIError, APIConnectionError, APIConnectionTimeoutError, APIUserAbortError, NotFoundError, ConflictError, RateLimitError, BadRequestError, AuthenticationError, InternalServerError, PermissionDeniedError, UnprocessableEntityError, } = Errors;
export var toFile = Uploads.toFile;
export var fileFromPath = Uploads.fileFromPath;
(function (LlamaStackClient) {
    LlamaStackClient.Telemetry = API.Telemetry;
    LlamaStackClient.Agents = API.Agents;
    LlamaStackClient.Datasets = API.Datasets;
    LlamaStackClient.Evaluate = API.Evaluate;
    LlamaStackClient.Evaluations = API.Evaluations;
    LlamaStackClient.Inference = API.Inference;
    LlamaStackClient.Safety = API.Safety;
    LlamaStackClient.Memory = API.Memory;
    LlamaStackClient.PostTraining = API.PostTraining;
    LlamaStackClient.RewardScoringResource = API.RewardScoringResource;
    LlamaStackClient.SyntheticDataGenerationResource = API.SyntheticDataGenerationResource;
    LlamaStackClient.BatchInference = API.BatchInference;
    LlamaStackClient.Models = API.Models;
    LlamaStackClient.MemoryBanks = API.MemoryBanks;
    LlamaStackClient.Shields = API.Shields;
})(LlamaStackClient || (LlamaStackClient = {}));
export default LlamaStackClient;
