// llama-stack-client/resources/post-training/jobs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Jobs = void 0;
const resource_1 = require("../../resource.js");
const core_1 = require("../../core.js");
class Jobs extends resource_1.APIResource {
    list(params = {}, options) {
        if ((0, core_1.isRequestOptions)(params)) {
            return this.list({}, params);
        }
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData } = params;
        return this._client.get('/post_training/jobs', {
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    artifacts(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...query } = params;
        return this._client.get('/post_training/job/artifacts', {
            query,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    cancel(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...body } = params;
        return this._client.post('/post_training/job/cancel', {
            body,
            ...options,
            headers: {
                Accept: '*/*',
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    logs(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...query } = params;
        return this._client.get('/post_training/job/logs', {
            query,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
    status(params, options) {
        const { 'X-LlamaStack-ProviderData': xLlamaStackProviderData, ...query } = params;
        return this._client.get('/post_training/job/status', {
            query,
            ...options,
            headers: {
                ...(xLlamaStackProviderData != null ?
                    { 'X-LlamaStack-ProviderData': xLlamaStackProviderData }
                    : undefined),
                ...options?.headers,
            },
        });
    }
}
exports.Jobs = Jobs;
(function (Jobs) {
})(Jobs = exports.Jobs || (exports.Jobs = {}));
